unitycatalog/functions/databricks.py

```diff
@@ -0,0 +1,549 @@
+
+if TYPE_CHECKING:
+    from databricks.sdk import WorkspaceClient
+    from databricks.sdk.service.catalog import (
+        ColumnTypeName,
+        CreateFunction,
+        FunctionInfo,
+        FunctionParameterInfo,
+    )
+    from databricks.sdk.service.sql import StatementParameterListItem
+
+EXECUTE_FUNCTION_ARG_NAME = "__execution_args__"
+DEFAULT_EXECUTE_FUNCTION_ARGS = {
+    "wait_timeout": "30s",
+    "row_limit": 100,
+    "byte_limit": 4096,
+}
+
+_logger = logging.getLogger(__name__)
+
+
+def get_default_databricks_workspace_client() -> "WorkspaceClient":
+    try:
+        from databricks.sdk import WorkspaceClient
+    except ImportError as e:
+        raise ImportError(
+            "Could not import databricks-sdk python package. "
+            "If you want to use databricks backend then "
+            "please install it with `pip install databricks-sdk`."
+        ) from e
+    return WorkspaceClient()
+
+
+def column_type_to_python_type(column_type: Enum) -> Any:
+    from databricks.sdk.service.catalog import ColumnTypeName
+
+    mapping = {
+        # numpy array is not accepted, it's not json serializable
+        ColumnTypeName.ARRAY: (list, tuple),
+        # a string expression in base64 format
+        ColumnTypeName.BINARY: str,
+        ColumnTypeName.BOOLEAN: bool,
+        # tinyint type
+        ColumnTypeName.BYTE: int,
+        ColumnTypeName.CHAR: str,
+        ColumnTypeName.DATE: (datetime.date, str),
+        # no precision and scale check, rely on SQL function to validate
+        ColumnTypeName.DECIMAL: (decimal.Decimal, float),
+        ColumnTypeName.DOUBLE: float,
+        ColumnTypeName.FLOAT: float,
+        ColumnTypeName.INT: int,
+        ColumnTypeName.INTERVAL: (datetime.timedelta, str),
+        ColumnTypeName.LONG: int,
+        ColumnTypeName.MAP: dict,
+        # ref: https://docs.databricks.com/en/error-messages/datatype-mismatch-error-class.html#null_type
+        # it's not supported in return data type as well `[UNSUPPORTED_DATATYPE] Unsupported data type "NULL". SQLSTATE: 0A000`
+        ColumnTypeName.NULL: type(None),
+        ColumnTypeName.SHORT: int,
+        ColumnTypeName.STRING: str,
+        ColumnTypeName.STRUCT: dict,
+        # not allowed for python udf, users should only pass string
+        ColumnTypeName.TABLE_TYPE: str,
+        ColumnTypeName.TIMESTAMP: (datetime.datetime, str),
+        ColumnTypeName.TIMESTAMP_NTZ: (datetime.datetime, str),
+        # it's a type that can be defined in scala, python shouldn't force check the type here
+        # ref: https://www.waitingforcode.com/apache-spark-sql/used-defined-type/read
+        ColumnTypeName.USER_DEFINED_TYPE: object,
+    }
+    if column_type not in mapping:
+        raise ValueError(f"Unsupported column type: {column_type}")
+    return mapping[column_type]
+
+
+def validate_param(param: Any, param_info: "FunctionParameterInfo"):
```

Comment:
Similar comment as above. It might be worthwhile to abstract this as a util that both Databricks and UC can use with conditional imports for the `ColumnTypeName` entries that are required for type validation. 
I like the idea of this module having only the client interface APIs in it.